### DISCUSSION TITLE
How to delete files and folders on formula uninstall?

### DISCUSSION THREAD

**Discussion by bleepnetworks** [Homebrew/discussions#567] [Tap maintenance and Homebrew development]
I have written a formula for a go-lang app, which uses **def post_install** to make a directory in /usr/local/share & copy data there.

How do I remove files and folders created during install, when the user uninstalls the formula?

You can do this with **uninstall delete: ["~/.blah.json"]** in casks, but I can't see an equivalent in formula.

**Comment by SMillerDev** [comment]
Anything under the `/usr/local/Cellar/formula/version` will be deleted and all files linking to it.

If you write outside of this it won't be deleted, there is no uninstall methods.

**Comment by gromgit** [comment]
To expand on Sean's point: if it's not something that's meant to be modified by users, stick it under `#{share}` and it'll be linked into `$HOMEBREW_PREFIX/share` on install _and_ cleaned up on uninstall. (If you haven't already read it, https://docs.brew.sh/Formula-Cookbook#variables-for-directory-locations is an important resource for formula authors.)

But if it's something that your users would be expected to customize, consider sticking it under `#{etc}`, then they will be warned about it on uninstall, and can then decide what to do with it:
```
$ brew uninstall httpd
Uninstalling /usr/local/Cellar/httpd/2.4.46_1... (1,624 files, 31.2MB)

Warning: The following httpd configuration files have not been removed!
If desired, remove them manually with `rm -rf`:
  /usr/local/etc/httpd
  /usr/local/etc/httpd/extra
  /usr/local/etc/httpd/extra/httpd-autoindex.conf
  /usr/local/etc/httpd/extra/httpd-dav.conf
  /usr/local/etc/httpd/extra/httpd-default.conf
  /usr/local/etc/httpd/extra/httpd-info.conf
  /usr/local/etc/httpd/extra/httpd-languages.conf
  /usr/local/etc/httpd/extra/httpd-manual.conf
  /usr/local/etc/httpd/extra/httpd-mpm.conf
  /usr/local/etc/httpd/extra/httpd-multilang-errordoc.conf
  /usr/local/etc/httpd/extra/httpd-ssl.conf
  /usr/local/etc/httpd/extra/httpd-userdir.conf
  /usr/local/etc/httpd/extra/httpd-vhosts.conf
  /usr/local/etc/httpd/extra/proxy-html.conf
  /usr/local/etc/httpd/httpd.conf
  /usr/local/etc/httpd/magic
  /usr/local/etc/httpd/mime.types
  /usr/local/etc/httpd/original
  /usr/local/etc/httpd/original/extra
  /usr/local/etc/httpd/original/extra/httpd-autoindex.conf
  /usr/local/etc/httpd/original/extra/httpd-dav.conf
  /usr/local/etc/httpd/original/extra/httpd-default.conf
  /usr/local/etc/httpd/original/extra/httpd-info.conf
  /usr/local/etc/httpd/original/extra/httpd-languages.conf
  /usr/local/etc/httpd/original/extra/httpd-manual.conf
  /usr/local/etc/httpd/original/extra/httpd-mpm.conf
  /usr/local/etc/httpd/original/extra/httpd-multilang-errordoc.conf
  /usr/local/etc/httpd/original/extra/httpd-ssl.conf
  /usr/local/etc/httpd/original/extra/httpd-userdir.conf
  /usr/local/etc/httpd/original/extra/httpd-vhosts.conf
  /usr/local/etc/httpd/original/extra/proxy-html.conf
  /usr/local/etc/httpd/original/httpd.conf
```
All of the above can be done in your `install` block, no post-install required.
